### PR TITLE
Re-implement select_object_content implementation

### PR DESCRIPTION
**Pull request by harshavardhana**
This change fixes multiple issues

- handles Unicode boundaries properly for special delimiters
- handle zero payloads 'Cont' event messages
- handle error messages properly

### CODE REVIEW

**Comment by harshavardhana** [comment]
PR is updated with further changes @Praveenrajmani @sinhaashish PTAL

**Comment by sinhaashish** [comment]
```

**Comment by harshavardhana** [comment]
Unicode characters should be inputs for python as `u'character'`

**Comment by harshavardhana** [comment]
```py
from minio import Minio
from minio.error import ResponseError

from minio.select.options import (SelectObjectOptions, CSVInput,
                                  JSONInput, RequestProgress,
                                  ParquetInput, InputSerialization,
                                  OutputSerialization, CSVOutput,
                                  JsonOutput)
from minio.select.errors import (SelectCRCValidationError, SelectMessageError)

client = Minio('s3.amazonaws.com',
               access_key='ACCESSKEY',
               secret_key='SECRETKEY')

options = SelectObjectOptions(
    expression="select * from s3object",
    input_serialization=InputSerialization(
        compression_type="GZIP",
        csv=CSVInput(FileHeaderInfo="USE",
                     RecordDelimiter="\n",
                     FieldDelimiter=u'╦',
                     QuoteCharacter='"',
                     QuoteEscapeCharacter='"',
                     Comments="#",
                     AllowQuotedRecordDelimiter="FALSE",
                     ),
        # If input is JSON
        # json=JSONInput(Type="DOCUMENT",)
        ),

    output_serialization=OutputSerialization(
        csv=CSVOutput(QuoteFields="ASNEEDED",
                      RecordDelimiter="\n",
                      FieldDelimiter=u'╦',
                      QuoteCharacter='"',
                      QuoteEscapeCharacter='"',)

        # json = JsonOutput(
        #     RecordDelimiter="\n",
        #     )
        ),
    request_progress=RequestProgress(
        enabled="False"
        )
    )

try:
    data = client.select_object_content('wlk-data-wbrp', '20190612-00690-1/wlk-wbrp-part-0000.csv.gz', options)

    # Get the records
    with open('my-record-file', 'w') as record_data:
        for d in data.stream(10*1024):
            record_data.write(d)

    # Get the stats
    print(data.stats())

except SelectMessageError as err:
    print(err)

except SelectCRCValidationError as err:
    print(err)

except ResponseError as err:
    print(err)
```

File: minio/select/reader.py [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+                    raise SelectMessageError(
+                        "Unrecognized content-type {0}".format(content_type))
+                else:
+                    payload_bytes = self.response.read(payload_length)
+                    self.stat = _parse_stats(payload_bytes)
+
+            elif event_type == EVENT_RECORDS:
+                payload_bytes = self.response.read(payload_length)
+        else:
+            raise SelectMessageError(
```
Comment (sinhaashish) [inline minio/select/reader.py]: We know the server response ,so do you think we can ever get this `unrecognised message type`

Comment (harshavardhana) [reply minio/select/reader.py]: Yes because server can prematurely, close the connection. Short reads are common.

**Pull request merged**